compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/AbstractParserErrorHandler.java

```diff
@@ -253,6 +254,8 @@ private Result seekMatch(ParserRuleContext currentCtx) {
                     "seekMatch caught " + exception;
             bestMatch = new Result(new ArrayDeque<>(), LOOKAHEAD_LIMIT - 1);
             bestMatch.solution = new Solution(Action.REMOVE, currentCtx, SyntaxKind.NONE, currentCtx.toString());
+        } finally {
+            this.ctxStack = tempCtxStack;
```

Comment:
Adding this line at the end of the catch block is enough. Since spotBugs fails with unused variable for L245 added as a finally block.